log-viewer/modules/TreeView.ts

```diff
@@ -492,6 +502,65 @@ function hideByDuration(elements: Array<HTMLElement>) {
   }
 }
 
+// Find the parent node of "elm" (or return null if we reach the root of the tree)
+function getParentNode(elm: Element | null) {
+  if (elm === null || elm.classList.contains('root')) {
+    return null;
+  }
+
+  let parent = elm.parentElement;
+  while (parent && !parent.classList.contains('node')) {
+    parent = parent.parentElement;
+  }
+
+  return parent;
+}
+
+function insertCrumb(container: Element, node: HTMLElement) {
+  // @ts-ignore (custom dom property)
+  const line = node.line as LogLine,
+    nameNode = node.querySelector<HTMLElement>('.name') || node,
+    crumb = document.createElement('div'),
+    textElm = document.createElement('div'),
+    textNode = document.createTextNode(line.getBreadcrumbText());
+
+  crumb.classList.add('crumb');
+  crumb.title = nameNode.textContent!;
+  crumb.appendChild(textElm);
+  crumb.addEventListener('click', evt => {
+    nameNode.scrollIntoView();
+    showBreadcrumb(nameNode);
+  });
+  textElm.appendChild(textNode);
+  container.insertAdjacentElement('afterbegin', crumb);
+}
+
+function showBreadcrumb(nameNode: HTMLElement | null) {
+  const container = document.getElementById('breadcrumb')!;
```

Comment:
Seems like this is not going to change? maybe pull it out of the function and do it once?

Reply:
The container does change (now that we are using replaceWith), but I have put it into a variable so that we can do it once.